File: day2/day2p1.py
file = open('input.txt', 'r+', encoding='utf-8')

lines = list()
line_array = list()

for line in file:
    lines.append(line.rstrip())

for line in lines:
    line_array.append(line.split('\t'))

file.close()

def process(line):
    temp = list()
    for num in line:
        temp.append(int(num))
    return max(temp) - min(temp)

total = 0
for line in line_array:
    total += process(line)

print(f'Checksum: {total}')

File: day2/day2p1.js
const fs = require('fs');

fs.readFile('./input.txt', 'utf-8', (err, data) => {
    let numArray = split(data);
    console.log(`Checksum: ${calculate(numArray)}`);
});

const split = (data) => {
    let lines = data.split('\n');
    let nums = [];

    lines.forEach(line => {
        nums.push(line.trim().split('\t'));
    });

    for(let i = 0; i < nums.length; i++){
        for(let j = 0; j < nums[i].length; j++){
            nums[i][j] = parseInt(nums[i][j]);
        }
    }
    return nums;
}

const calculate = (nums) => {
    let rowResult = 0;

    nums.forEach(row => {
        rowResult += Math.max(...row) - Math.min(...row);
    });
    return rowResult;
}